wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -168,9 +204,40 @@ class WireJsonAdapterFactory private constructor(
       }
     }.nullSafe()
 
+    internal val UINT64_STRING_JSON_ADAPTER = object : JsonAdapter<Long>() {
+      @Throws(IOException::class)
+      override fun fromJson(reader: JsonReader): Long? {
+        return when (reader.peek()) {
+          JsonReader.Token.STRING -> {
+            UINT64_JSON_ADAPTER.fromJson(reader)
+          }
+          else -> {
+            reader.nextLong()
+          }
+        }
+      }
+
+      @Throws(IOException::class)
+      override fun toJson(writer: JsonWriter, value: Long?) {
+        writer.value(UINT64_JSON_ADAPTER.toJsonValue(value).toString())
+      }
+    }.nullSafe()
+
+    internal val SINT64_STRING_JSON_ADAPTER = object : JsonAdapter<Long>() {
+      @Throws(IOException::class)
+      override fun fromJson(reader: JsonReader): Long? {
+        return UINT64_STRING_JSON_ADAPTER.fromJson(reader)
```

Comment:
I don’t think this makes sense; it’ll permit negative values

Reply:
Tests are working both with `Long.MIN_VALUE` and `Long.MAX_VALUE`, confirmed by protoc tests.

Reply:
LGTM